### ISSUE TITLE
"use v6.*" no longer works

### ISSUE THREAD

**Issue by timo** [rakudo/rakudo#2952]
## The Problem

A recent change made "use v6.*" give an exception.

## Expected Behavior

Probably want it to select the highest language version that is not `.PREVIEW`.

## Actual Behavior

===SORRY!=== Error while compiling -e
No compiler available for Perl v6.*
at -e:1
------> use v6.*⏏<EOL>

**Comment by timo** [comment]
impacted modules

| File | Code |
|--|--|
| [finanalyst/pod-cached<br>``…/PrecompilationDoc.pm6`` :*1*:](https://github.com/finanalyst/pod-cached/blob/5ec09b9320e4b98edcf068def75c2218c82692c3/lib/PrecompilationDoc.pm6#L1) | <code><b>use v6.\*</b>;</code> |
| [finanalyst/pod-cached<br>``…/100-heavy-duty.t`` :*2*:](https://github.com/finanalyst/pod-cached/blob/5ec09b9320e4b98edcf068def75c2218c82692c3/t/100-heavy-duty.t#L2) | <code><b>use v6.\*</b>;</code> |
| [jonathanstowe/Audio-Liquidsoap<br>``…/010-use.t`` :*3*:](https://github.com/jonathanstowe/Audio-Liquidsoap/blob/4c2f108deb1be9ce08f0b040d5d2b007b7f2a77b/t/010-use.t#L3) | <code><b>use v6.\*</b>;</code> |
| [jonathanstowe/Audio-Liquidsoap<br>``…/030-builtins.t`` :*3*:](https://github.com/jonathanstowe/Audio-Liquidsoap/blob/4c2f108deb1be9ce08f0b040d5d2b007b7f2a77b/t/030-builtins.t#L3) | <code><b>use v6.\*</b>;</code> |
| [jonathanstowe/Audio-Liquidsoap<br>``…/050-list.t`` :*3*:](https://github.com/jonathanstowe/Audio-Liquidsoap/blob/4c2f108deb1be9ce08f0b040d5d2b007b7f2a77b/t/050-list.t#L3) | <code><b>use v6.\*</b>;</code> |
| [tadzik/JSON-Unmarshal<br>``…/null.t`` :*3*:](https://github.com/tadzik/JSON-Unmarshal/blob/e1b6288c5f3165058f36c0f4e171cdf2dfd640da/t/null.t#L3) | <code><b>use v6.\*</b>;</code> |

**Comment by ugexe** [comment]
Note that in all prior versions `use v6.*` would load `v6.c` even though no `use` at all would load `v6.d` (so all current usage of `use v6.*` is used under broken assumptions)

```
$ perl6 -e 'use v6.*; say $*PERL'
Perl 6 (6.c)
$ perl6 -e 'use v6.d; say $*PERL'
Perl 6 (6.d)
```

**Comment by vrurg** [comment]
Fixed by #2953. Moreover, it loads the highest version available and any form supported by `Version` works. Simply put, `v6.*`, `v6.c+` would now load `6.d`.

A controversial part is `v6.*.*` which would load `6.e.PREVIEW`. But getting around this would require additional efforts and I see no sense of doing it. The only use case for three-part form I see for now is to load a specific modifier for test purposes. For example, `6.d` currently has `TEST` and `TESTDEPR` modifiers used in `t/02-rakudo/14-revisions.t`. Those could as well be loaded with `v6.*.TEST`.

In either case, all this functionality is just a side effect of `Version` matching mechanism. 

I wonder if it makes sense to have this in a test? It seems like we only need it for backward compatibility purpose.

**Comment by vrurg** [comment]
Closed via #2953
